src/plugins/compression.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Compression Library Wrapper
+ *
+ *  Created on: 08 Nov 2018
+ *      Author: Oliver Fernandes
+ */
+
+#ifndef SRC_PLUGINS_COMPRESSION_H_
+#define SRC_PLUGINS_COMPRESSION_H_
+
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <vector>
+
+#ifdef ENABLE_LZ4
+#include "lz4.h"
+#endif
+
+using ByteIterator = std::vector<char>::iterator;
```

Comment:
not entirely certain, whether this `using` directive should be here.
`using namespace` should not occur within header files (at least if they are unscoped). In this case, it is not as bad, but should probably still be avoided

Reply:
maybe make it class internal

Reply:
Absolutely. Moved it into the class.

Btw., it's not a using-declaration, it's a type alias (no namespaces involved).
https://en.cppreference.com/w/cpp/language/type_alias